crates/oro-diagnostics/src/lib.rs

```diff
@@ -0,0 +1,14 @@
+pub trait Diagnostic: std::error::Error {
+    fn code(&self) -> &DiagnosticCode;
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+#[non_exhaustive]
+pub enum DiagnosticCode {
+    /// Failed to parse a package spec.
```

Comment:
It feels a bit redundant to have to duplicate the description: once in a comment here and once in  a place where the error is defined. Might become out of sync? I couldn't come up with a quick alternative though...

Reply:
But the direction LGTM!

Reply:
Yeah, the point of duplicating the message is so you can just go to `docs.rs/oro-diagnostics` and look up every single diagnostic code without having to look into the source code for orogene or find which library a particular diagnostic is coming from. The duplication is unfortunate, but they're not necessarily exactly the same message.

I'm a bit frustrated by it, too, but I'm trying to think of a better way and having a hard time doing so.